### PR TITLE
[#1731] Add Hovering documentation to the Eclipse Support website.

### PR DESCRIPTION
**Pull request by miklossy**
Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

### CODE REVIEW

File: xtext-website/documentation/310_eclipse_support.md [hunk 1]
```diff
@@ -514,6 +514,34 @@ Often the default strategy only needs some guidance (e.g. selecting the text cor
 
 The hyperlinks are provided by the [HyperlinkHelper]({{site.src.xtext_eclipse}}/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hyperlinking/HyperlinkHelper.java) which will create links for cross-referenced objects by default. Clients may want to override `createHyperlinksByOffset(XtextResource, int, IHyperlinkAcceptor)` to provide additional links or supersede the default implementation.
 
+## Hovering {#hovering}
+Similar to [hyperlinking](#hyperlinking), Xtext-based editors provide hovering support on tokens corresponding to cross-references out-of-the-box. When hovering over a cross-reference token, the Xtext framework shows the documentation of the element the cross-reference is referring to. Considering the Xtext Simple Arithmetics example, when hovering over e.g. a function call, a popup window displays the documentation of the called function:
```
Comment (cdietrich) [inline xtext-website/documentation/310_eclipse_support.md:518]: actually you con hover at everything that represents a name. (declarations + references)

File: xtext-website/documentation/310_eclipse_support.md [hunk 2]
```diff
@@ -514,6 +514,34 @@ Often the default strategy only needs some guidance (e.g. selecting the text cor
...
+
+![](images/hovering.png)
+
+The default hovering behaviour can be customized by extending the [DefaultEObjectHoverProvider]({{site.src.xtext_eclipse}}/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/html/DefaultEObjectHoverProvider.java) class, overriding the `getHoverInfoAsHtml(EObject o)` method and binding the custom implementation in the corresponding UI module:
```
Comment (cdietrich) [inline xtext-website/documentation/310_eclipse_support.md:522]: also hint about IEObjectDocumentationProvider / MultiLineCommentDocumentationProvider

File: xtext-website/documentation/310_eclipse_support.md (hunk 1 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md]: done

File: xtext-website/documentation/310_eclipse_support.md (hunk 2 above)
Comment (miklossy) [reply xtext-website/documentation/310_eclipse_support.md]: done

**Pull request merged**